tf2/include/tf2/time.h

```diff
@@ -52,8 +54,32 @@ namespace tf2
     return std::chrono::system_clock::now();
   }
 
-  inline double durationToSec(const tf2::Duration & input){
-    return (double)std::chrono::duration_cast<std::chrono::seconds>(input).count();
+  inline Duration durationFromSec(double t_sec)
+  {
+    int32_t sec, nsec;
+    sec = (int32_t)floor(t_sec);
+    nsec = (int32_t)std::nearbyint((t_sec-sec) * 1e9);
```

Comment:
Btw, this isn't in our style guide, but it is in the Google Style guide which ours is based on: https://google.github.io/styleguide/cppguide.html#Casting

When writing our style guide I tried to avoid copying the entire Google Style guide, so I don't know if this can be made more visible or not.

I also do not think cpplint.py checks this (it may not be possible to do so easily).